### PR TITLE
fix: 4497 use Translation for Recipe created

### PR DESCRIPTION
**Pull request by itsrubberduck**
## What type of PR is this?
- bug

## What this PR does / why we need it:
This PR fixes untranslated timeline events in the recipe creation process. Currently, when creating a new recipe, the timeline event shows a hardcoded English message "Recipe Created" regardless of the user's selected language.

Changes made:
- Changed hardcoded string in `recipe_service.py` to use translation key
- Added initial translations for:
  - de-DE: "Rezept erstellt"
  - en-US: "Recipe created"

This is the first step in fully localizing timeline events, starting with the two main languages. Additional language support will be added in follow-up PRs.

## Which issue(s) this PR fixes:
Partial fix for #4497

## Special notes for your reviewer:
The translation key follows the existing pattern used in recipe defaults:
- Current: `recipe.recipe-defaults.ingredient-note`
- New: `recipe.recipe-created`

## Testing
Tested by:
1. Creating a new recipe with German (de-DE) locale
2. Verifying timeline event shows "Rezept erstellt"
3. Creating a new recipe with English (en-US) locale
4. Verifying timeline event shows "Recipe created"

All existing tests pass. Translation keys are consistent with existing naming patterns.

### CODE REVIEW

**Comment by Kuchenpirat** [comment]
Hey, thanks for your PR.
It seems like you created this PR on top of your other one.

**Comment by Kuchenpirat** [comment]
Also you only need to add base translations to the en_US file. The rest is handled through crowdin. You can add the german translation over there as soon as the PR including the translation is merged.

**Comment by itsrubberduck** [comment]
Thank you! I hope it's correct now.

File: mealie/lang/messages/en-US.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
     },
     "recipe": {
         "unique-name-error": "Recipe names must be unique",
+        "recipe-created": "Recipe created",
```
Comment (michael-genson) [inline mealie/lang/messages/en-US.json]: Should this be "Recipe Created"? (vs "Recipe created")

**Review by michael-genson — CHANGES_REQUESTED** [review]
Thanks for this! One small thing

File: mealie/lang/messages/en-US.json (hunk 1 above)
Comment (itsrubberduck) [reply mealie/lang/messages/en-US.json]: Fixed it

**Review by michael-genson — APPROVED** [review]
Thank you!

**Comment by boc-the-git** [comment]
Please provide a meaningful title to the pull request.

**Pull request merged**